victor/src/main/groovy/com/trello/victor/RasterizeTask.groovy

```diff
@@ -84,35 +93,70 @@ class RasterizeTask extends DefaultTask {
-            File resDir = getResourceDir(density)
-            resDir.mkdirs()
+        if (generateVectorDrawables) {
+            createOutput()
+        } else {
+            includeDensities.each { Density density ->
+                createOutput(density)
+            }
         }
 
-        Converter converter = new Converter()
-        svgFiles.each { File svgFile ->
-            SVGResource svgResource = new SVGResource(svgFile, baseDpi)
+        if (generateVectorDrawables) {
+            svgFiles.each { File svgFile ->
+                File resDir = getResourceDir()
+                File destination = new File(resDir, getDestinationFile(svgFile.name))
+                OutputStream outStream = new FileOutputStream(destination)
+                String error = Svg2Vector.parseSvgToXml(svgFile, outStream)
+                if (!error.isEmpty()) {
+                    logger.error(error)
+                    throw new GradleException(error)
+                }
+
+                outStream.flush()
+                outStream.close()
 
-            includeDensities.each { Density density ->
-                File destination = new File(getResourceDir(density), getDestinationFile(svgFile.name))
-                converter.transcode(svgResource, density, destination)
                 logger.info("Converted $svgFile to $destination")
             }
+        } else {
+            Converter converter = new Converter()
+            svgFiles.each { File svgFile ->
+                SVGResource svgResource = new SVGResource(svgFile, baseDpi)
+
+                includeDensities.each { Density density ->
+                    File destination = new File(getResourceDir(density), getDestinationFile(svgFile.name))
+                    converter.transcode(svgResource, density, destination)
+                    logger.info("Converted $svgFile to $destination")
+                }
+            }
         }
 
         inputs.removed { change ->
             logger.debug("$change.file.name was removed; removing it from generated folder")
 
-            includeDensities.each { Density density ->
-                File resDir = getResourceDir(density)
-                File file = new File(resDir, getDestinationFile(change.file.name))
-                file.delete()
+            if (generateVectorDrawables) {
+                cleanupInput()
+            } else {
+                includeDensities.each { Density density ->
+                    cleanupInput(density)
+                }
             }
         }
     }
 
-    File getResourceDir(Density density) {
-        return new File(outputDir, "/drawable-${density.name().toLowerCase()}")
+    void createOutput(@Nullable Density density = null) {
+        File resDir = getResourceDir(density)
+        resDir.mkdirs()
+    }
+
+    void cleanupInput(@Nullable Density density = null) {
+        File resDir = getResourceDir(density)
+        File file = new File(resDir, getDestinationFile(change.file.name))
```

Comment:
That's a very good question. What's worrying me is that I'm sure I tested it. I'll triple check again..  Sorry about that.